### ISSUE TITLE
Feature request - noisy neighbour detection

### ISSUE THREAD

**Issue by grzesuav** [coroot/coroot#358]
Recently I came across this https://netflixtechblog.com/noisy-neighbor-detection-with-ebpf-64b1f4b3bbdd - which described how eBPF can be used to detect noisy neighours. Would be great to have something similar in coroot

**Comment by def** [comment]
It's already there: 
 * https://coroot.com/blog/engineering/linux-delay-accounting/
 * CPU Delay: https://coroot.com/docs/coroot/inspections/cpu

**Comment by grzesuav** [comment]
it it possible to see on the node level the highest offenders ? I mean which pod is causing delay to others ? On my side we are thinking of identifying most offending workloads with the aim to evict them from the node

**Comment by def** [comment]
CPU delay shows if a container is not getting enough CPU time. This can happen due to CPU throttling or the node running out of capacity. In that case, you can sort CPU consumers by their CPU usage.

**Comment by def** [comment]
<img width="1197" alt="Screenshot 2024-10-16 at 14 04 17" src="https://github.com/user-attachments/assets/f2de18b2-e6fa-4c6f-981f-33bd13efdc41">

**Comment by grzesuav** [comment]
@def But usage is not enough, this consumption can be lower than request. I would say the noisy neighbour is the pod exceeding requests, so just showing consumers is not enough IMO

**Comment by def** [comment]
IMO, usage is the correct metric here because, regardless of the CPU request, each process consumes CPU time, making cores temporarily unavailable for other computations.

**Comment by grzesuav** [comment]
I agree, however this is similar like with MemoryPressure and DiskPressure, idea which I have in mind is to detect and evict the offending workload
